lab_1/main.py

```diff
@@ -40,7 +83,39 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
+
+    if not isinstance(freq_dict, dict) or not isinstance(top_n, int):
+        return None
+
+    words = len(freq_dict)
+    freq_val = reversed(sorted(freq_dict.values()))
+    sorted_freq = {}
+
+    for i in freq_val:
+        for k in freq_dict.keys():
+            if freq_dict[k] == i:
+                sorted_freq[k] = freq_dict[k]
+
+    if top_n > words:
+        top_words = sorted_freq
+        return top_words
+    if top_n < words:
+        top_words = list(sorted_freq.keys())[:(top_n)]
+        return top_words
+
+
+
+freq_dict_en = calculate_frequencies(tokens_en)
+freq_dict_de = calculate_frequencies(tokens_de)
+freq_dict_unknown = calculate_frequencies(tokens_unknown)
+
+new_dicts_en = {v: k for k, v in freq_dict_en.items()}
+new_dicts_de = {v: k for k, v in freq_dict_de.items()}
+new_dicts_unknown = {v: k for k, v in freq_dict_unknown.items()}
+
+top_n_en = len(new_dicts_en)
+top_n_de = len(new_dicts_de)
+top_n_unknown = len(new_dicts_unknown)
```

Comment:
The `main.py` file should not contain function calls - you should check your implementation using the `start.py` file.